controller.py

```diff
@@ -13,6 +14,8 @@ def __init__(self, networkManager: Optional[any] = None):
     self.networkManager = networkManager
     self.variables = {}
     self.actions = {}
+    t = threading.Thread.__init__(self)
```

Comment:
When creating the thread variable, `t`, it is being created as a local variable, and it gets destroyed after the constructor is done. Also, you would need to pass the run function as an argument to thread:
`self.thread = threading.Thread(target=self.run)`
Before starting the thread, we should also make a member variable to decide if the thread should keep running, and set it to true: `self.running = True`